### ISSUE TITLE
Issues related to evaluating the compression effect

### ISSUE THREAD

**Issue by tang12138-yyhh** [eml-eda/mixprec-pruning#7]
For example, case: icl_mixprec_training_cost_size_time.py

After your method, I can obtain the log information, which includes the model structure information (resnet-8), the parameter update information during the training process (Tensor), the size of the model (tensor.size), and the saved model after the final fine-tuning (.pt).

But I have the following problems:

I don't know how to evaluate the pruning and quantization effects of the model, such as the pruning rate and which channels are pruned?

2. The quantized model should have undergone layer renaming (with the information names of the quantized part added), which makes it impossible for me to make inferences (tests) based on this quantized information in many places?

3. How can I view the parameters of the quantized model, such as the quantized weight values or pruning effects? Because I tried some methods in python but failed to identify the layer information of the model, resulting in the extraction failure.

**Comment by bmots** [comment]
Hi @tang12138-yyhh ,
1) you can see in the log a "final architectural summary", followed by a dictionary with the layer names as keys, and the activation and weight precision as value. In the `w_precision` field, you will see a list of integers, one per weight channel. The pruned channels are the ones which have a precision of 0 (with the idea that pruning can be seen as quantizing to 0 bits).

2) The inference procedure is exactly as before the conversion and the quantization-aware training, i.e. calling the model on the input data. Could you make an example of the problem? 

3) Given for instance the layer `inputblock.conv1` of `exported_model`, you can see the selected weight precision for each channel with `exported_model.seed.inputblock.conv1.selected_w_precision`. You can check the quantized weight values for that layer with:

    ```python
    exported_model.eval()
    for qtz in exported_model.seed.inputblock.conv1.mixprec_w_quantizer.mix_qtz:
        qtz.dequantize = False  # This will show you in the next lines the integer values of the weights, as no dequantization operation will be applied in the forward function

    exported_model.seed.inputblock.conv1.mixprec_w_quantizer(exported_model.seed.inputblock.conv1.weight)
    ```
    This simply applies the weight quantizer of that layer on the weight values.
